src/prefect/experimental/workers/base.py

```diff
@@ -0,0 +1,613 @@
+        await self.scan_storage_for_deployments()
+        # schedule the scheduled flow run polling loop to run every `query_seconds`
+        self._loop_task_group.start_soon(
+            partial(
+                critical_service_loop,
+                workload=self._get_and_submit_flow_runs,
+                interval=self._query_seconds,
+                printer=self._logger.debug,
+            )
+        )
+        # schedule the sync loop to run every `heartbeat_seconds`
+        self._loop_task_group.start_soon(
+            partial(
+                critical_service_loop,
+                workload=self.sync_worker,
+                interval=self._heartbeat_seconds,
+                printer=self._logger.debug,
+            )
+        )
+        # schedule the storage scan loop to run
+        self._loop_task_group.start_soon(
+            partial(
+                critical_service_loop,
+                workload=self.scan_storage_for_deployments,
+                interval=self._workflow_storage_scan_seconds,
+                printer=self._logger.debug,
+            )
+        )
+
+    async def _get_and_submit_flow_runs(self):
+        # if the pool is paused or has a 0 concurrency limit, don't bother polling
+        if self._worker_pool and (
+            self._worker_pool.is_paused or self._worker_pool.concurrency_limit == 0
+        ):
+            return
+
+        runs_response = await self._get_scheduled_flow_runs()
+        await self._submit_scheduled_flow_runs(flow_run_response=runs_response)
+
+    async def _update_local_worker_pool_info(self):
+        try:
+            worker_pool = await self._client.read_worker_pool(
+                worker_pool_name=self._worker_pool_name
+            )
+        except ObjectNotFound:
+            if self._create_pool_if_not_found:
+                worker_pool = await self._client.create_worker_pool(
+                    worker_pool=schemas.actions.WorkerPoolCreate(
+                        name=self._worker_pool_name, type=self.type
+                    )
+                )
+                self._logger.info(f"Worker pool {self._worker_pool_name!r} created.")
+            else:
+                self._logger.warning(
+                    f"Worker pool {self._worker_pool_name!r} not found!"
+                )
+                return
+
+        # if the remote config type changes (or if it's being loaded for the
+        # first time), check if it matches the local type and warn if not
+        if getattr(self._worker_pool, "type", 0) != worker_pool.type:
+            if worker_pool.type != self.__class__.type:
+                self._logger.warning(
+                    f"Worker type mismatch! This worker process expects type "
+                    f"{self.type!r} but received {worker_pool.type!r}"
+                    " from the server. Unexpected behavior may occur."
+                )
+        self._worker_pool = worker_pool
+
+    async def _update_local_queue_list(self):
```

Comment:
I vote to remove this and rely on the API to keep things up to date for now until we have a need to synchronize queue lists between clients / servers; as discussed this probably relates to my comment on the get schedule runs method on the client as well